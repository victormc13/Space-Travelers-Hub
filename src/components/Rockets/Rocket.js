import PropTypes from 'prop-types';

function Rocket({ rocket, onReserve }) {
  const badgeStyle = rocket.reserved
    ? { padding: '2px 10px' }
    : { padding: '0' };

  const buttonStyle = rocket.reserved
    ? { background: '#fff', color: '#000' }
    : { background: '#007bff', border: 'none' };

  return (
    <div className="rocket">
      <img src={rocket.flickr_images[0]} alt={rocket.name} className="rocket-img" />
      <div className="rocket-detail">
        <h2>{rocket.name}</h2>
        <p>
          <span style={badgeStyle} className="badge">
            {rocket.reserved ? 'Reserved' : ''}
          </span>
          {rocket.description}
        </p>
        <button
          style={buttonStyle}
          className="btn-reservation"
          type="button"
          onClick={() => onReserve(rocket.id)}
        >
          {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
        </button>
      </div>
    </div>
  );
}

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
    name: PropTypes.string.isRequired,
    reserved: PropTypes.bool.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
  onReserve: PropTypes.func.isRequired,
};

export default Rocket;
