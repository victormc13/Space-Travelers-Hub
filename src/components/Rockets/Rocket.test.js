import React from 'react';
import Rocket from './Rocket';

const { render, fireEvent } = require('@testing-library/react');

test('Rocket component renders correctly', () => {
  const rocket = {
    id: 'rocket-1',
    flickr_images: ['image-url'],
    name: 'Falcon 9',
    reserved: false,
    description: 'Sample rocket description',
  };
  const onReserve = jest.fn();

  const { container, getByText } = render(<Rocket rocket={rocket} onReserve={onReserve} />);

  expect(container).toMatchSnapshot();

  expect(getByText('Falcon 9')).toBeInTheDocument();
  expect(getByText('Sample rocket description')).toBeInTheDocument();

  fireEvent.click(getByText('Reserve Rocket'));
  expect(onReserve).toHaveBeenCalledWith('rocket-1');
});
