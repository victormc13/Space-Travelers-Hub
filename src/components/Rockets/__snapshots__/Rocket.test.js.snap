// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component renders correctly 1`] = `
<div>
  <div
    class="rocket"
  >
    <img
      alt="Falcon 9"
      class="rocket-img"
      src="image-url"
    />
    <div
      class="rocket-detail"
    >
      <h2>
        Falcon 9
      </h2>
      <p>
        <span
          class="badge"
          style="padding: 0px;"
        />
        Sample rocket description
      </p>
      <button
        class="btn-reservation"
        style="background: rgb(0, 123, 255);"
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
